napi.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+var log = require('npmlog')
+var path = require('path')
+var pkg = require(path.resolve('package.json'))
+
+var versionArray = process.version
+  .substr(1)
+  .replace(/-.*$/, '')
+  .split('.')
+  .map(function (item) {
+    return +item
+  })
+
+module.exports.isNapiRuntime = function (runtime) {
+  return runtime === 'napi'
+}
+
+module.exports.isSupportedVersion = function (napiVerison) {
+  var version = parseInt(napiVerison, 10)
+  return version <= getNapiVersion() && packageSupportsVersion(version)
+}
+
+var packageSupportsVersion = function (napiVerison) {
+  if (pkg.binary && pkg.binary.napi_versions) {
+    for (var i = 0; i < pkg.binary.napi_versions.length; i++) {
+      if (pkg.binary.napi_versions[i] === napiVerison) return true
+    };
+  };
+  return false
+}
+
+module.exports.warnUnsupportedVersion = function (v) {
```

Comment:
The code goes on to attempt the download which fails because in nearly all cases the prebuilt binary does not exist. However, if there was some sort of misconfiguration be between `prebuild`, `prebuild-install`, and the `package.json`, a workable binary _might_ exist. Do you prefer it stopped with an instead?